packages/providers/src/core/PoapCollectionsApi/PoapCollectionsApi.ts

```diff
@@ -0,0 +1,148 @@
+   */
+  public async createCollection(
+    createCollectionInput: CreateCollectionInput,
+    accessToken: string,
+  ): Promise<CollectionResponse> {
+    return await this.secureFetch<CollectionResponse>(
+      `${this.baseUrl}/collections`,
+      {
+        method: 'POST',
+        body: JSON.stringify(createCollectionInput),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      },
+      accessToken,
+    );
+  }
+
+  /**
+   * Applies a update to an existing collection in the POAP Collections API.
+   *
+   * @public
+   * @async
+   * @param {UpdateCollectionInput} updateCollectionInput - The new collection data to be updated.
+   * @param {string} accessToken - The access token for authorization.
+   * @returns {Promise<CollectionResponse>} A promise that resolves with the updated collection response from the API.
+   */
+  public async updateCollection(
+    updateCollectionInput: UpdateCollectionInput,
+    accessToken: string,
+  ): Promise<CollectionResponse> {
+    return await this.secureFetch<CollectionResponse>(
+      `${this.baseUrl}/collections/${updateCollectionInput.collectionId}`,
+      {
+        method: 'PUT',
+        body: JSON.stringify(updateCollectionInput),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      },
+      accessToken,
+    );
+  }
+
+  /**
+   * Sends a secure HTTP request to the POAP Collections API.
+   *
+   * This method is used internally by `createCollection` and `patchCollection` to send HTTP requests.
+   *
+   * @private
+   * @async
+   * @param {string} url - The URL for the HTTP request.
+   * @param {RequestInit} options - The options for the HTTP request, including method and body.
+   * @param {string} accessToken - The access token for authorization.
+   * @returns {Promise<R>} A promise that resolves with the response from the API.
+   */
+  private async secureFetch<R>(
+    url: string,
+    options: RequestInit,
+    accessToken: string,
+  ): Promise<R> {
+    const headersWithApiKey = {
+      ...options.headers,
+      Authorization: `Bearer ${accessToken}`,
+    };
+
+    const response = await fetch(url, {
+      method: options.method,
+      body: options.body,
+      headers: headersWithApiKey,
+    });
+
+    this.handleResponseStatus(response);
+
+    return await response.json();
+  }
+
+  /**
+   * Handles HTTP status codes and throws corresponding errors.
+   *
+   * @private
+   * @function
+   * @name PPoapCollectionsApi#handleHttpStatus
+   * @param {Response} response - The response from the fetch call.
+   * @throws {CollectionsUnauthorizedError} for 401 Unauthorized status codes.
+   * @throws {CollectionsBadRequestError} for 400 Bad Request status codes.
+   * @throws {Error} for other non-200 status codes.
+   */
+  private handleResponseStatus(response: Response): void {
+    switch (response.status) {
+      case 400:
+        throw new CollectionsBadRequestError();
```

Comment:
I don't know how Collections API works, but this could extract the message from the response body if any. How else will the user know that some input was incorrect?